### PR TITLE
Add NVIDIA tools for GPU support in containers

### PR DESCRIPTION
**Pull request by arnaldo2792**
**Issue number:**
N / A

**Description of changes:**

Related to #1799 , that PR is quite big. I'm taking some of the commits in that PR as their individual PR. The packages in this PR were build in #1799, but this PR also updates the packages to their latest versions.

```
* bb4a5a30 packages: add nvidia-container-toolkit
* 24b3812a packages: add libnvidia-container
```

**Testing done:**

I included both commits in a local branch based on #1799, and I launched a container with k8s. I was able to execute `nvidia-smi` from the container:

```
~ ❯ kubectl exec cuda-vectoradd -- nvidia-smi
Wed Jan 12 02:42:10 2022
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 470.82.01    Driver Version: 470.82.01    CUDA Version: 11.2     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  Tesla V100-SXM2...  Off  | 00000000:00:1E.0 Off |                    0 |
| N/A   32C    P0    25W / 300W |      0MiB / 16160MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|  No running processes found                                                 |
+-----------------------------------------------------------------------------+

```

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

File: packages/libnvidia-container/libnvidia-container.spec [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+export WITH_LIBELF=yes \\\
+export WITH_SECCOMP=yes \\\
+export WITH_TIRPC=yes \\\
+export prefix=%{_cross_prefix} \\\
+export DESTDIR=%{buildroot} \\\
+%{nil}
+
+%build
+%set_env
+make
```
Comment (bcressey) [inline packages/libnvidia-container/libnvidia-container.spec]: Can we use `%make_build` here?

File: packages/nvidia-container-toolkit/nvidia-container-toolkit.spec [hunk 2]
```diff
@@ -0,0 +1,56 @@
+%global debug_package %{nil}
```
Comment (bcressey) [inline packages/nvidia-container-toolkit/nvidia-container-toolkit.spec]: You shouldn't need this.

File: packages/nvidia-container-toolkit/nvidia-container-toolkit.spec [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+%description
+%{summary}.
+
+%prep
+%autosetup -n %{gorepo}-%{gover} -p1
+%cross_go_setup %{gorepo}-%{gover} %{goproject} %{goimport}
+
+%build
+%cross_go_configure %{goimport}
+go build -buildmode=pie -ldflags=-linkmode=external -o nvidia-container-toolkit ./cmd/nvidia-container-toolkit
```
Comment (bcressey) [inline packages/nvidia-container-toolkit/nvidia-container-toolkit.spec]: This should fix the rpm build failure that turning off the debug package fixed:

```suggestion
go build -buildmode=pie -ldflags="${GOLDFLAGS}" -o nvidia-container-toolkit ./cmd/nvidia-container-toolkit
```

File: packages/nvidia-container-toolkit/nvidia-container-toolkit.spec [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+install -d %{buildroot}%{_cross_bindir}
+install -d %{buildroot}%{_cross_tmpfilesdir}
+install -d %{buildroot}%{_cross_templatedir}
+install -d %{buildroot}%{_cross_datadir}/nvidia-container-toolkit
+install -d %{buildroot}%{_cross_factorydir}/etc/nvidia-container-runtime
+install -p -m 0755 nvidia-container-toolkit %{buildroot}%{_cross_bindir}/nvidia-container-runtime-hook
+install -m 0644 %{S:1} %{buildroot}%{_cross_factorydir}/etc/nvidia-container-runtime/config.toml
+install -m 0644 %{S:2} %{buildroot}%{_cross_tmpfilesdir}/nvidia-container-toolkit.conf
+install -m 0644 %{S:3} %{buildroot}%{_cross_templatedir}/nvidia-oci-hooks-json
+ln -s ../bin/shimpei %{buildroot}%{_cross_bindir}/nvidia-oci
```
Comment (bcressey) [inline packages/nvidia-container-toolkit/nvidia-container-toolkit.spec]:
```suggestion
ln -s shimpei %{buildroot}%{_cross_bindir}/nvidia-oci
```

**Comment by arnaldo2792** [comment]
Forced push includes:

- Use `%make_build`
- Fix go ld flags
- Fix symlink to `shimpei`

**Review by bcressey — APPROVED** [review]

File: packages/libnvidia-container/Cargo.toml [hunk 5]
```diff
@@ -0,0 +1,23 @@
+[package]
+name = "libnvidia-container"
+version = "0.1.0"
+edition = "2018"
+publish = false
+build = "build.rs"
+
+[lib]
+path = "pkg.rs"
+
```
Comment (jpculp) [inline packages/libnvidia-container/Cargo.toml:10]:
```suggestion

[package.metadata.build-package]
releases-url = "https://github.com/NVIDIA/libnvidia-container/releases"

```

We started including the `releases-url` with packages.

File: packages/nvidia-container-toolkit/Cargo.toml [hunk 6]
```diff
@@ -0,0 +1,20 @@
+[package]
+name = "nvidia-container-toolkit"
+version = "0.1.0"
+edition = "2018"
+publish = false
+build = "build.rs"
+
+[lib]
+path = "pkg.rs"
+
```
Comment (jpculp) [inline packages/nvidia-container-toolkit/Cargo.toml:10]:
```suggestion

[package.metadata.build-package]
releases-url = "https://github.com/NVIDIA/nvidia-container-toolkit/releases"

```
same as above

**Review by etungsten — APPROVED** [review]

**Comment by arnaldo2792** [comment]
Forced push includes `releases-url` in both packages

**Review by jpculp — APPROVED** [review]

**Pull request merged**